clusterctl/clusterdeployer/clusterclient.go

```diff
@@ -65,15 +67,16 @@ func (c *clusterClient) removeKubeconfigFile() error {
 	return os.Remove(c.kubeconfigFile)
 }
 
-func NewClusterClientFromFile(kubeconfigFile string) (*clusterClient, error) {
-	c, err := clientcmd.NewClusterApiClientForDefaultSearchPath(kubeconfigFile)
+func NewClusterClientFromDefaultSearchPath(kubeconfigFile string, overrides tcmd.ConfigOverrides) (*clusterClient, error) {
```

Comment:
This one treats the first param as a string and the prior treats it as the actual config? That seems backwards to me, but maybe that's because the libraries you are calling need to do the file handling....